packages/flutter/lib/src/widgets/animated_switcher.dart

```diff
@@ -292,39 +325,37 @@ class _AnimatedSwitcherState extends State<AnimatedSwitcher> with TickerProvider
     if (_currentChild != null) {
       _currentChild.controller.dispose();
     }
-    for (_AnimatedSwitcherChildEntry child in _children) {
+    for (_AnimatedSwitcherChildEntry child in _previousChildren) {
       child.controller.dispose();
     }
     super.dispose();
   }
 
-  bool get hasNewChild => widget.child != null;
-  bool get hasOldChild => _currentChild != null;
-
   @override
   void didUpdateWidget(AnimatedSwitcher oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (hasNewChild != hasOldChild || hasNewChild &&
-        !Widget.canUpdate(widget.child, _currentChild.widgetChild)) {
+    final bool hasNewChild = widget.child != null;
+    final bool hasOldChild = _currentChild != null;
+    if (hasNewChild != hasOldChild ||
+        hasNewChild && !Widget.canUpdate(widget.child, _currentChild.widgetChild)) {
       _addEntry(animate: true);
     } else {
+      // Make sure we update the child widget in _currentChild even if we're not
+      // going to start a new animation, but keep the key from the previous
+      // transition, since otherwise there is a huge performance penalty.
```

Comment:
s/since otherwise there is a huge performance penalty//. If you don't keep the key, then you're not going to update the child, you're going to replace it.

Reply:
Done